misk/src/main/kotlin/misk/web/actions/reflect.kt

```diff
@@ -0,0 +1,60 @@
+package misk.web.actions
+
+import java.lang.reflect.Method
+import java.util.ArrayDeque
+import kotlin.reflect.KFunction
+import kotlin.reflect.jvm.javaMethod
+
+/** Returns an instance of [T] annotating this method or a method it overrides. */
+internal inline fun <reified T : Annotation> KFunction<*>.findAnnotationWithOverrides(): T? {
+  return javaMethod!!.findAnnotationWithOverrides(T::class.java)
+}
+
+/** Returns an instance of [T] annotating this method or a method it overrides. */
```

Comment:
Maybe mention that if the method overrides a method from multiple superclasses all with the same annotation, it returns one of these "arbitrarily".

Reply:
Thanks! Fixed.